src/spring-cloud/azext_spring_cloud/_params.py

```diff
@@ -74,8 +74,11 @@ def load_arguments(self, _):
         c.argument('name', name_type, help='Name of app.')
 
     with self.argument_context('spring-cloud app create') as c:
-        c.argument(
-            'is_public', arg_type=get_three_state_flag(), help='If true, assign public domain', default=False)
+        c.argument('assign_endpoint', arg_type=get_three_state_flag(),
+                   help='If true, assign endpoint URL for direct access.', default=False,
+                   options_list=['--assign-endpoint', c.deprecate(target='--is-public', redirect='--assign-endpoint')])
+        c.argument('is_public', arg_type=get_three_state_flag(),
+                   deprecate_info=c.deprecate(expiration='2.2.0'), help='Renamed since 2.1.3. Please use assign-endpoint instead.')
```

Comment:
```suggestion
        c.argument('assign_endpoint', arg_type=get_three_state_flag(),
                   help='If true, assign endpoint URL for direct access.', default=False,
                   options_list=['--assign-endpoint', c.deprecate(target='--is-public', redirect='--assign-endpoint')])
```
The second line should be removed as `is_public` is removed in the `custom.py` function and it will not be shown in help message. You already put `--is-public` as a deprecated option of `assign_endpoint`.

You may also consider adding `hide=True` for `--is-public`, then it will not be shown in help message (reduce confusion for new users who never used this deprecated option) but there will be a deprecation warning when users still use it (not break existing users who use it but make them aware of the deprecation).